discretize/tensor_mesh.py

```diff
@@ -159,6 +162,53 @@ def _repr_html_(self):
         fmt += "</table>\n"
         return fmt
 
+    def __iter__(self):
+        """
+        Iterator over the cells
+        """
+        if self.dim == 1:
+            (nx,) = self.shape_cells
+            indices = ((i,) for i in range(nx))
+        if self.dim == 2:
+            nx, ny = self.shape_cells
+            indices = ((i, j) for j in range(ny) for i in range(nx))
+        elif self.dim == 3:
+            nx, ny, nz = self.shape_cells
+            indices = (
+                (i, j, k) for k in range(nz) for j in range(ny) for i in range(nx)
+            )
+        iterator = (self[index] for index in indices)
+        return iterator
+
+    def __getitem__(self, indices):
```

Comment:
I think it should accept both linear indices, `[0, n_cells)` and tuples, `[ix, iy, iz]`. I would also rely on numpy's `ravel_multi_index` function reliably correctly convert from the tuple to the linear index.

Reply:
Done! By allowing `__getitem__` to accept integer indices, I could greatly simplify the `__iter__`.